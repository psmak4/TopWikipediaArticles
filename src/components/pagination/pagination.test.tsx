/* eslint-disable @typescript-eslint/no-unused-vars */
import { fireEvent, render, screen } from '@testing-library/react'
import Article from '~/models/article'
import ArticlesStore, { setCurrentPage } from '~/stores/articlesStore'
import Pagination from './pagination'

jest.mock('~/stores/articlesStore')
const mockArticlesStore = ArticlesStore as jest.MockedFunction<typeof ArticlesStore>
const mockSetCurrentPage = setCurrentPage as jest.MockedFunction<typeof setCurrentPage>
const mockCurrentPage = 1
const mockPageSize = 1
const mockArticles: Article[] = [
	{
		article: 'one',
		views: 1,
		rank: 1,
	},
	{
		article: 'two',
		views: 2,
		rank: 2,
	},
	{
		article: 'three',
		views: 3,
		rank: 3,
	},
]

describe('Pagination', () => {
	beforeEach(() => {
		mockArticlesStore.mockImplementation(() => ({
			articles: mockArticles,
			currentPage: mockCurrentPage,
			pageSize: mockPageSize,
			pins: new Map<string, Article>(),
			searchDate: new Date(new Date().setDate(new Date().getDate() - 1)),
		}))
		mockSetCurrentPage.mockImplementation(_ => void 0)
	})

	afterEach(() => {
		jest.clearAllMocks()
	})

	test('renders pagination buttons', () => {
		render(<Pagination />)

		// Check if the Pagination component renders the expected number of buttons
		const pageButtons = screen.getAllByRole('button')
		expect(pageButtons).toHaveLength(5) // prev, 1, 2, 3, next

		// Check if the buttons are rendered correctly
		for (let i = 1; i < pageButtons.length - 1; i++) {
			const button = pageButtons[i]
			expect(button).toHaveTextContent(i.toString())
		}

		// Check if the current page button is disabled
		const currentPageButton = pageButtons[1]
		expect(currentPageButton).toHaveAttribute('disabled')
	})

	test('calls setCurrentPage when a page button is clicked', () => {
		render(<Pagination />)

		// Click on a page button
		const pageButton = screen.getByText('2')
		fireEvent.click(pageButton)

		// Check if setCurrentPage is called with the correct argument
		expect(mockSetCurrentPage).toHaveBeenCalledWith(2)
	})

	test('matches snapshot', () => {
		const { asFragment } = render(<Pagination />)
		expect(asFragment()).toMatchSnapshot()
	})
})
