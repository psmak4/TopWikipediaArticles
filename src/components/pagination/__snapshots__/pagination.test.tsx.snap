// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <button
      class="directionButton"
    >
      <span
        class="icon"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.2339 13C10.4444 13 10.6316 12.933 10.7719 12.799C11.076 12.531 11.076 12.062 10.7719 11.794L6.81871 7.99721L10.7719 4.22278C11.076 3.95477 11.076 3.48576 10.7719 3.21776C10.4912 2.92741 10 2.92741 9.7193 3.21776L5.22807 7.50586C4.92398 7.77387 4.92398 8.24288 5.22807 8.51089L9.7193 12.799C9.85965 12.933 10.0468 13 10.2339 13Z"
          />
        </svg>
      </span>
    </button>
    <div
      class="buttonContainer"
    >
      <button
        class="pageButton pageButtonSelected"
        disabled=""
      >
        <span
          class="buttonContent"
        >
          1
        </span>
      </button>
      <button
        class="pageButton"
      >
        <span
          class="buttonContent"
        >
          2
        </span>
      </button>
      <button
        class="pageButton"
      >
        <span
          class="buttonContent"
        >
          3
        </span>
      </button>
    </div>
    <button
      class="directionButton"
    >
      <span
        class="icon"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.76608 13C5.55556 13 5.36842 12.933 5.22807 12.799C4.92398 12.531 4.92398 12.062 5.22807 11.794L9.18129 7.99721L5.22807 4.22278C4.92398 3.95477 4.92398 3.48576 5.22807 3.21776C5.50877 2.92741 6 2.92741 6.2807 3.21776L10.7719 7.50586C11.076 7.77387 11.076 8.24288 10.7719 8.51089L6.2807 12.799C6.14035 12.933 5.95322 13 5.76608 13Z"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
